sklego/preprocessing/pandastransformers.py

```diff
@@ -43,16 +44,22 @@ def _nw_select_dtypes(df, include: str | list[str], exclude: str | list[str]):
     if isinstance(exclude, str):
         exclude = [exclude]
 
-    include = include or ["string", "number", "bool", "category"]
```

Comment:
I don't think this logic was right to begin with. If `include` isn't passed, it should keep everything except for what's in `exclude` - not just the string/number/bool/category cols not in `exclude`...
I've added a test which covers this